msl/equipment/resources/vaisala/ptu300.py

```diff
@@ -0,0 +1,199 @@
+"""
+Vaisala Barometer which reads temperature, relative humidity, and pressure, e.g. of type PTU300
+"""
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from msl.equipment import EquipmentRecord
+from msl.equipment.resources import register
+from msl.equipment.exceptions import VaisalaError
+from msl.equipment.connection_serial import ConnectionSerial
+
+
+@register(manufacturer=r'Vaisala', model=r'PTU30*', flags=re.IGNORECASE)
+class PTU300(ConnectionSerial):
+
+    def __init__(self, record: EquipmentRecord) -> None:
+        """Vaisala Barometer PTU300 series.
+        Device manual: https://docs.vaisala.com/v/u/M210855EN-D/en-US
+
+        Do not instantiate this class directly. Use the :meth:`~.EquipmentRecord.connect`
+        method to connect to the equipment.
+
+        :param record: A record from an :ref:`equipment-database`.
+        """
+        super(PTU300, self).__init__(record)
+
+        self.set_exception_class(VaisalaError)
+        self.rstrip = True
+
+        self.ID = record.serial
+
+        self.stop_run_mode()
+
+        self._units = {}
+        self.pressure_modules = set()
+        self.info = self.device_info(show=False)
+        self.check_serial()
+
+    def check_serial(self) -> str:
+        """Get the device ID (serial) number and check it agrees with the equipment record.
+
+        :return: the serial number as a string
+        """
+        # can use reply = self.query("*9900SN") but the serial number is also in the device info
+        sn = self.info["Serial number"]
+        if not sn == self.ID:
+            self.raise_exception(f"Inconsistent serial number: expected {self.ID} but received {sn}")
+        return sn
+
+    def device_info(self, show: bool = True) -> dict:
+        """Prints information about the Vaisala device.
+
+        :param show: bool for whether to display the device info as log messages
+        :return: a dictionary of the device information
+        """
+        info = {}
+        break_keys = {
+            'PTU30': "module 2",
+            'PTB33': "module 4",
+        }
+        break_key = "TBC"
+        self.write("?")
+        while True:
+            ok = self.read()
+            if show:
+                self.log_info(ok)
```

Comment:
I suggest that using the logging module within `msl-equipment` is only intended for `msl-equipment` developers for debugging purposes and not try to anticipate what the end user may want to see displayed in the console or written to a file.

For example, I can call this method with `show=True` and still not see the logging message. This depends on how someone configures the logging level.